frame/dex-router/src/lib.rs

```diff
@@ -159,23 +305,26 @@ pub mod pallet {
 			asset_pair: CurrencyPair<T::AssetId>,
 			route: BoundedVec<T::PoolId, T::MaxHopsInRoute>,
 		) -> Result<(), DispatchError> {
-			let k1 = asset_pair.base;
-			let k2 = asset_pair.quote;
-			Self::check_route(asset_pair, &route)?;
-			let existing_route = DexRoutes::<T>::get(k1, k2);
-			DexRoutes::<T>::insert(k1, k2, DexRoute::Direct(route.clone()));
+			Self::validate_route(asset_pair, &route)?;
+			let base_asset = asset_pair.base;
+			let quote_asset = asset_pair.quote;
+			let existing_route = Self::get_route(asset_pair);
+			// remove any route for base_asset/quote_asset or symmetric quote_asset/base_asset pair.
+			DexRoutes::<T>::remove(base_asset, quote_asset);
+			DexRoutes::<T>::remove(quote_asset, base_asset);
+			DexRoutes::<T>::insert(base_asset, quote_asset, DexRoute::Direct(route.clone()));
```

Comment:
I don't really understand why we try both way, can't we index by the currency pair and sort it before inserting/removing/indexing on DexRoutes? Also, unless I am missing something, you are not inserting the inverse pair here so in the end it won't work if the user is querying the swapped pair?

Reply:
if we store sorted pair then to make any operation on pool of route, we have to compare if pool's pair and operation's pair are same or pair needed to be swapped() to make correct operation. 
Instead of above I use same pair as of pool as key in storage and in get_route() method check if request pair is same or reversed and based on this while executing operation on pool use appropriate pair. 
Please check 
```
  2         /// Returns pair of route and bool to indicate if route should be used in reverse direction                                                                                                                                      
  1         /// with assets swapped.                                                                                                                                                                                                         
378         fn get_route(asset_pair: CurrencyPair<T::AssetId>) -> Option<(Vec<T::PoolId>, bool)> {  
```
And also check tests works for both way operations. 
I have also simplified confusing code (on which you have requested changes).

Reply:
Alright :+1: